crates/polars-plan/src/dsl/mod.rs

```diff
@@ -1150,9 +1150,6 @@ impl Expr {
     /// Should be used in aggregation context. If you want to filter on a
     /// DataFrame level, use `LazyFrame::filter`.
     pub fn filter<E: Into<Expr>>(self, predicate: E) -> Self {
```

Comment:
We should check this in `dsl` conversion and either raise an error or translate to a `Filter` node.